src/common/hooks/useEndorsementChain/fetchTokenTransfer.ts

```diff
@@ -0,0 +1,70 @@
+import { TradeTrustToken } from "@govtechsg/token-registry/contracts";
+import { TypedEvent } from "@govtechsg/token-registry/dist/contracts/common";
+import { LogDescription } from "ethers/lib/utils";
+import { TokenTransferEvent, TokenTransferEventType } from "../../../types";
+import { sortLogChain } from "./helpers";
+
+export const fetchTokenTransfers = async (
+  tokenRegistry: TradeTrustToken,
+  tokenId: string
+): Promise<TokenTransferEvent[]> => {
+  // Fetch transfer logs from token registry
+  const tokenRegistryAddress = tokenRegistry.address;
+  const identifyTokenTransferEvents = identifyTokenTransferEventsFunction(tokenRegistryAddress);
+  const transferLogFilter = tokenRegistry.filters.Transfer(null, null, tokenId);
+  const logs = await tokenRegistry.queryFilter(transferLogFilter, 0);
+  if (logs.length === 0) {
+    throw new Error("Unminted Title Escrow");
+  }
+  const formattedLogs = logs.map((log) => {
+    if (!log.args) throw new Error(`Transfer log malformed: ${log}`);
+    if (!log.blockNumber) throw new Error("Block number not present");
+    if (!log.transactionHash) throw new Error("Transaction hash not present");
+    return {
+      ...log,
+      ...tokenRegistry.interface.parseLog(log),
+    };
+  });
```

Comment:
i see similarity of this helper with `getParsedLogs`, if can reuse else leave it